services/gateway/gateway-service.mjs

```diff
@@ -310,24 +346,22 @@ class GatewayService {
     return true;
   }
 
-  _getDistanceSquare(a,b){
-    return Math.pow(Math.abs(a.x - b.x),2) + Math.pow(Math.abs(a.y - b.y),2);
+  _getManhattanDistance(a,b){
```

Comment:
I think this should be a utility function in `map.mjs` instead of a member function of `GatewayService`.

Reply:
I'll leave this to whyang.